### PR TITLE
Make localhost uploads from worker to webui faster

### PR DESCRIPTION
**Pull request by kraih**
When the `LOCAL_UPLOAD=1` setting is present in the `workers.ini`, the
worker will check if the webui is on `localhost`/`127.0.0.1` and then
copy assets directly on the file system instead of uploading them
via HTTP. In benchmarks uploading a 200MB asset (20 chunks) this
has been about 27 times faster than before.

If the feature proves stable in production it might be worth
removing the `LOCAL_UPLOAD` setting in the future and relying only
on host detection. We might need to add an environment variable
instead however, to ensure full stack tests stay full stack and
don't end up using only this optimization.

### CODE REVIEW

File: lib/OpenQA/Client/Handler.pm [hunk 1]
```diff
@@ -39,5 +39,11 @@ sub _build_url {
 
 sub _build_post { $_[0]->client->build_tx(POST => shift()->_build_url(+shift()) => form => +shift()) }
 
+sub is_local {
+    my $self = shift;
+    my $host = $self->_build_url('/')->to_abs->host;
+    return $host eq 'localhost' || $host eq '127.0.0.1';
```
Comment (aaannz) [inline lib/OpenQA/Client/Handler.pm:45]: Is IPv6 supported? Shouldn't here be '::1' as well?

Comment (kraih) [reply lib/OpenQA/Client/Handler.pm]: And added.

**Review by Martchus — APPROVED** [review]

**Comment by psladek** [comment]
Have tested this patch on suma retail server image builds, very significant improvement, ~50GB image is now copied in ~2minutes instead of hour and half.

**Review by perlpunk — APPROVED** [review]

**Comment by Martchus** [comment]
Unfortunately that doesn't speak for the efficiency of our HTTP file transfer.

**Comment by kraih** [comment]
The HTTP upload mechanism is definitely much less efficient than it could be. The data is copied around more than it has to be... but the code is also very complex and involves modules like `OpenQA::Parser::Result` (for some reason), so i didn't dare trying to make quick optimizations. Any attempts will have to be done very carefully, and would be an "Epic" ticket for sure.

**Pull request merged**